test/e2e/info_test.go

```diff
@@ -202,11 +205,22 @@ var _ = Describe("Podman Info", func() {
 		}
 
 		for _, tt := range backends {
+			oldDesiredDB := os.Getenv(desiredDB)
+			if tt.arg == "boltdb" {
+				err := os.Setenv(desiredDB, "boltdb")
```

Comment:
I'm really uncomfortable with this: if one of the `Expect()s` below fails, it'll leave the envariable set, and subsequent tests will fail in distracting/redherring ways. Would it be possible to
```
     const desiredDB ...
+    oldDB = os.Getenv()
+    defer { restore oldDB }
```
Way up there instead of here?

Reply:
We'd have to retain the existing cleanup code as that handles it on a per-loop basis not a per-function basis but we can definitely add the defer in addition, it shouldn't hurt even if cleanup was otherwise successful.